diff -ruN fw.orig/etc/fstab fw.patched/etc/fstab
--- fw.orig/etc/fstab	2018-01-09 11:17:52.124855211 +0100
+++ fw.patched/etc/fstab	2018-01-09 11:22:24.330632351 +0100
@@ -1,13 +1,3 @@
 # /etc/fstab: static file system information.
 #
 # <file system> <mount pt>     <type>   <options>         <dump> <pass>
-/dev/root       /              ext2     rw,noauto         0      1
-proc            /proc          proc     defaults          0      0
-devpts          /dev/pts       devpts   defaults,gid=5,mode=620   0      0
-tmpfs           /dev/shm       tmpfs    mode=0777         0      0
-tmpfs           /tmp           tmpfs    defaults          0      0
-mediafs         /media         tmpfs    defaults          0      0
-sysfs           /sys           sysfs    defaults          0      0
-varfs           /var           tmpfs    defaults,size=196M 0      0
-debugfs 	/sys/kernel/debug debugfs 		  0 	0
-ubi1:user	/usr/local	ubifs	defaults,noatime,rw,sync	0 0
diff -ruN fw.orig/etc/init.d/S00eQ3SystemStart fw.patched/etc/init.d/S00eQ3SystemStart
--- fw.orig/etc/init.d/S00eQ3SystemStart	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S00eQ3SystemStart	2018-01-09 11:22:24.330632351 +0100
@@ -7,17 +7,6 @@
 SERIAL=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
 
 init() {
-	# power led on
-	echo 255 > /sys/class/leds/power/brightness
-	echo default-on > /sys/class/leds/power/trigger
-	# internet led off
-	echo none > /sys/class/leds/internet/trigger
-	# info led fast
-	echo timer > /sys/class/leds/info/trigger
-	echo 255 > /sys/class/leds/info/brightness
-	echo 100 > /sys/class/leds/info/delay_off
-	echo 100 > /sys/class/leds/info/delay_on
-
         chmod 775 /var
         mkdir /var/log
         chmod 775 /var/log
@@ -73,27 +62,6 @@
 	if [ ! -e /usr/local/etc/config/shadow ] ; then
 		cp $CFG_TEMPLATE_DIR/shadow /usr/local/etc/config/shadow
 	fi
-
-	modprobe spidev
-        modprobe ic200_spi
-        modprobe spi_eq3_gpio
-        modprobe spi_bitbang
-        mknod spidev0.0 c 153 0
-        modprobe gpio-keys
-        modprobe fsl_usb2_udc.ko
-	if grep -q "eQ3Mode=production" /proc/cmdline ; then
-          modprobe g_multi luns=1 cdrom=1 ro=1 removable=0,0 file=/usr/share/eq3/install.iso host_addr=00:1a:22:00:05:86 dev_addr=00:1a:22:00:05:87 iManufacturer="eQ-3" iProduct="CCU2" idVendor="0x1b1f" idProduct="0xc016" iSerialNumber="$SERIAL"
-	fi
-	
-	# Tunneling
-        modprobe tun
-       
-	# USB
-        modprobe ehci-hcd
-
-	# HM/HmIP Dual Protocol
-	modprobe mxs_raw_auart
-	modprobe eq3_char_loop
 	 
 	if [ ! -d /usr/local/etc/config ] ; then
                 mkdir -p /usr/local/etc/config
@@ -129,7 +97,7 @@
 
 stop () {
 	start-stop-daemon -K -q -p /var/run/crond.pid
-	/bin/update_firmware_pre
+       /etc/piVCCU/install_pending_addon.sh
 }
 
 restart() {
diff -ruN fw.orig/etc/init.d/S10udev fw.patched/etc/init.d/S10udev
--- fw.orig/etc/init.d/S10udev	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S10udev	1970-01-01 01:00:00.000000000 +0100
@@ -1,48 +0,0 @@
-#!/bin/sh
-#
-# udev	This is a minimal non-LSB version of a UDEV startup script.  It
-#	was derived by stripping down the udev-058 LSB version for use
-#	with buildroot on embedded hardware using Linux 2.6.34+ kernels.
-#
-#	You may need to customize this for your system's resource limits
-#	(including startup time!) and administration.  For example, if
-#	your early userspace has a custom initramfs or initrd you might
-#	need /dev much earlier; or without hotpluggable busses (like USB,
-#	PCMCIA, MMC/SD, and so on) your /dev might be static after boot.
-#
-#	This script assumes your system boots right into the eventual root
-#	filesystem, and that init runs this udev script before any programs
-#	needing more device nodes than the bare-bones set -- /dev/console,
-#	/dev/zero, /dev/null -- that's needed to boot and run this script.
-#
-
-# Check for missing binaries
-UDEV_BIN=/lib/udev/udevd
-test -x $UDEV_BIN || exit 5
-
-# Check for config file and read it
-UDEV_CONFIG=/etc/udev/udev.conf
-test -r $UDEV_CONFIG || exit 6
-. $UDEV_CONFIG
-
-case "$1" in
-    start)
-        echo -n "Populating ${udev_root:-/dev} using udev: "
-        echo -e '\000\000\000\000' > /proc/sys/kernel/hotplug
-        $UDEV_BIN -d || (echo "FAIL" && exit 1)
-        udevadm trigger
-        echo "done"
-        ;;
-    stop)
-        # Stop execution of events
-        udevadm control --stop-exec-queue
-        killall udevd
-        ;;
-    *)
-        echo "Usage: $0 {start|stop}"
-        exit 1
-        ;;
-esac
-
-
-exit 0
diff -ruN fw.orig/etc/init.d/S15watchdog fw.patched/etc/init.d/S15watchdog
--- fw.orig/etc/init.d/S15watchdog	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S15watchdog	1970-01-01 01:00:00.000000000 +0100
@@ -1,20 +0,0 @@
-#!/bin/sh
-#
-# Start watchdog
-#
-
-case "$1" in
- start)
-	echo "Starting watchdog..."
-	watchdog -t 30 /dev/watchdog
-	;;
-  stop)
-	;;
-  restart|reload)
-	;;
-  *)
-	echo $"Usage: $0 {start|stop|restart}"
-	exit 1
-esac
-
-exit $?
diff -ruN fw.orig/etc/init.d/S20urandom fw.patched/etc/init.d/S20urandom
--- fw.orig/etc/init.d/S20urandom	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S20urandom	1970-01-01 01:00:00.000000000 +0100
@@ -1,54 +0,0 @@
-#! /bin/sh
-#
-# urandom	This script saves the random seed between reboots.
-#		It is called from the boot, halt and reboot scripts.
-#
-# Version:	@(#)urandom  1.33  22-Jun-1998  miquels@cistron.nl
-#
-
-[ -c /dev/urandom ] || exit 0
-#. /etc/default/rcS
-
-case "$1" in
-	start|"")
-		if [ "$VERBOSE" != no ]
-		then
-			echo -n "Initializing random number generator... "
-		fi
-		# Load and then save 512 bytes,
-		# which is the size of the entropy pool
-		if [ -f /etc/random-seed ]
-		then
-			cat /etc/random-seed >/dev/urandom
-		fi
-		# check for read only file system
-		if ! touch /etc/random-seed 2>/dev/null
-		then
-			echo "read-only file system detected...done"
-			exit
-		fi
-		rm -f /etc/random-seed
-		umask 077
-		dd if=/dev/urandom of=/etc/random-seed count=1 \
-			>/dev/null 2>&1 || echo "urandom start: failed."
-		umask 022
-		[ "$VERBOSE" != no ] && echo "done."
-		;;
-	stop)
-		if ! touch /etc/random-seed 2>/dev/null
-                then
-                        exit
-                fi
-		# Carry a random seed from shut-down to start-up;
-		# see documentation in linux/drivers/char/random.c
-		[ "$VERBOSE" != no ] && echo -n "Saving random seed... "
-		umask 077
-		dd if=/dev/urandom of=/etc/random-seed count=1 \
-			>/dev/null 2>&1 || echo "urandom stop: failed."
-		[ "$VERBOSE" != no ] && echo "done."
-		;;
-	*)
-		echo "Usage: urandom {start|stop}" >&2
-		exit 1
-		;;
-esac
diff -ruN fw.orig/etc/init.d/S40UsbNetwork fw.patched/etc/init.d/S40UsbNetwork
--- fw.orig/etc/init.d/S40UsbNetwork	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S40UsbNetwork	1970-01-01 01:00:00.000000000 +0100
@@ -1,52 +0,0 @@
-#!/bin/sh
-#
-# Starts Ethernet over USB
-#
-
-
-start() {
-	export TZ=`cat /etc/config/TZ`
-
-	echo -n "Starting network over usb: "
-		
-	if [ ! -f /etc/udhcpd.usb0.conf ]; then
-		echo "ERROR (file /etc/udhcp.usb0.conf not found)"
-	elif [ ! -f /etc/dnsd.conf ]; then
-		echo "ERROR (file /etc/dnsd.conf not found)"
-	else
-		ifconfig usb0 10.101.82.51 netmask 255.255.255.0
-		udhcpd -S /etc/udhcpd.usb0.conf
-		# disable DNS Server; CCU can only be access by ip address
-		#dnsd -d -c /etc/dnsd.conf -i 10.101.82.51
-		echo "OK"	
-	fi		
-}
-
-stop() {
-	echo -n "Stopping network over usb: "
-	killall udhcpd
-	killall dnsd	
-	echo "OK"
-}
-restart() {
-	stop
-	start
-}
-
-case "$1" in
-  start)
-	start
-	;;
-  stop)
-	stop
-	;;
-  restart|reload)
-	restart
-	;;
-  *)
-	echo "Usage: $0 {start|stop|restart}"
-	exit 1
-esac
-
-exit $?
-
diff -ruN fw.orig/etc/init.d/S50eq3configd fw.patched/etc/init.d/S50eq3configd
--- fw.orig/etc/init.d/S50eq3configd	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S50eq3configd	2018-01-09 11:22:24.330632351 +0100
@@ -17,6 +17,7 @@
 }
 
 start() {
+       sleep 5 # Wait some time to get the network interface complettly up
 	echo -n "Starting eq3configd: "
 	init
 	start-stop-daemon -S -q -p /var/run/eq3configd.pid --exec /bin/eq3configd &
diff -ruN fw.orig/etc/init.d/S60multimacd fw.patched/etc/init.d/S60multimacd
--- fw.orig/etc/init.d/S60multimacd	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S60multimacd	2018-01-09 11:22:24.330632351 +0100
@@ -10,11 +10,11 @@
 update() {        
 	if [ ! -e /etc/config/no-coprocessor-update ] ; then
 		echo "checking if firmware update is needed..."
-		/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u
+		/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u -t HM-MOD-UART
 		if ! [ $? -eq 0 ] ; then
 			echo "error while updating coprocessor, recovering..."
 			avrprog --bo
-			/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u -f
+			/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u -f -t HM-MOD-UART
 			sleep 1
 		else
 			echo "done"
diff -ruN fw.orig/etc/init.d/S99eQ3SystemStarted fw.patched/etc/init.d/S99eQ3SystemStarted
--- fw.orig/etc/init.d/S99eQ3SystemStarted	2018-01-09 11:17:52.120855185 +0100
+++ fw.patched/etc/init.d/S99eQ3SystemStarted	2018-01-09 11:22:24.330632351 +0100
@@ -8,15 +8,6 @@
 
 init() {
 	export TZ=`cat /etc/config/TZ`
-
-	# power led on
-	echo 255 > /sys/class/leds/power/brightness
-	echo default-on > /sys/class/leds/power/trigger
-	# internet led off
-	echo none > /sys/class/leds/internet/trigger
-	# info led off
-	echo none > /sys/class/leds/info/trigger
-	/bin/hss_led 2>&1 > /dev/null  &
 }
 
 start() {
diff -ruN fw.orig/etc/inittab fw.patched/etc/inittab
--- fw.orig/etc/inittab	2018-01-09 11:17:52.124855211 +0100
+++ fw.patched/etc/inittab	2018-01-09 11:22:24.330632351 +0100
@@ -14,25 +14,15 @@
 # process   == program to run
 
 # Startup the system
-null::sysinit:/bin/mount -t proc proc /proc
 #null::sysinit:/bin/mount -o remount,rw / # REMOUNT_ROOTFS_RW
-null::sysinit:/bin/mkdir -p /dev/pts
-null::sysinit:/bin/mkdir -p /dev/shm
 null::sysinit:/bin/mount -a
 null::sysinit:/bin/hostname -F /etc/hostname
 # now run any rc scripts
 ::sysinit:/etc/init.d/rcS
 
-# Put a getty on the serial port
-ttyAMA0::respawn:/sbin/getty -L ttyAMA0 115200 vt100 # GENERIC_SERIAL
-ttyGS0::respawn:/sbin/getty -L ttyGS0 115200 vt100 # USB SERIAL
-
 # Stuff to do for the 3-finger salute
 ::ctrlaltdel:/sbin/reboot
 
-# Stuff to do for SIGQUIT 
-::restart:/bin/update_firmware_run
-
 # Stuff to do before rebooting
 ::shutdown:/etc/init.d/rcK
 null::shutdown:/bin/umount -a -r
diff -ruN fw.orig/firmware/fwmap fw.patched/firmware/fwmap
--- fw.orig/firmware/fwmap	2018-01-09 11:17:52.948860587 +0100
+++ fw.patched/firmware/fwmap	2018-01-09 11:22:24.330632351 +0100
@@ -17,7 +17,8 @@
 
 #Coprozessor
 ARM7                    hss_comm.enc                                    0.056
-CCU2                    coprocessor_update.eq3                          2.8.6          # Dual CoProzessor CCU2  
+#CCU2                    coprocessor_update.eq3                          2.8.6          # Dual CoProzessor CCU2  
+CCU2                    dualcopro_si1002_update_blhm.eq3                2.8.6          # Dual CoProzessor CCU2  
 # CCU2                    coprocessor_update_hm_only.eq3                1.4.1          # HM only CoProzessor CCU2  
 CCU2-RFLGW				coprocessor_update_hm_only.eq3					1.4.1          # HM only CoProzessor CCU2 for HomeMatic RF Lan Gateway
 
diff -ruN fw.orig/opt/HMServer/pages/StorageSettingsDialog.ftl fw.patched/opt/HMServer/pages/StorageSettingsDialog.ftl
--- fw.orig/opt/HMServer/pages/StorageSettingsDialog.ftl	2018-01-09 11:17:52.192855654 +0100
+++ fw.patched/opt/HMServer/pages/StorageSettingsDialog.ftl	2018-01-09 11:22:24.330632351 +0100
@@ -168,23 +168,6 @@
 						<td class="CLASS21112">${"$"}{dialogSettingsSDCardStatus}:</td>
 						<td>${SDCardStatus}</td>
 					</tr>
-					<tr>
-						<td align="center" class="CLASS21112" colspan="2" >
-							<div class="popupControls CLASS21107">
-								<table>
-								<tr>
-									<td align="right">
-										<div id="initaliseSDCardButton" class="StdButton CLASS04907" onClick="OnInitaliseSDCard()">${"$"}{dialogSettingsStorageSettingsBtnInitaliseSDCard}</div>
-									</td>
-									<td align="right">
-										<div id="backupSDCardButton" class="StdButton CLASS04907" onClick="OnBackupSDCard()">${"$"}{dialogSettingsStorageSettingsBtnBackup}</div>
-									</td>
-								</tr>
-								</table>
-							</div>
-								<!--<div id="ejectSDCardButton" class="StdButton" onClick="OnEjectSDCard()">${"$"}{dialogSettingsStorageSettingsBtnEjectSDCard}</div> -->
-						</td>
-					</tr>
 				</table>
 			</td>
 			<td class="CLASS21113" align="left">
diff -ruN fw.orig/www/config/control_panel.cgi fw.patched/www/config/control_panel.cgi
--- fw.orig/www/config/control_panel.cgi	2018-01-09 11:17:52.752859308 +0100
+++ fw.patched/www/config/control_panel.cgi	2018-01-09 11:22:24.330632351 +0100
@@ -183,9 +183,12 @@
 
 if { "[read_var /etc/config/tweaks CP_DEVCONFIG]" != "" } {
   puts "<!-- devconfig -->"
-  puts "<td>"
-  puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">devconfig</div>"
-  puts "</td><td class=\"StdTableBtnHelp\"></td>"
+    puts "<td>"
+      puts "<div class=\"cpButton\">"
+      puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">DevConfig</div>"
+      puts "<div class=\"StdTableBtnHelp\"></div>"
+    puts "</div>"
+  puts "</td>"
   incr i
 }
 
diff -ruN fw.orig/www/config/cp_maintenance.cgi fw.patched/www/config/cp_maintenance.cgi
--- fw.orig/www/config/cp_maintenance.cgi	2018-01-09 11:17:52.752859308 +0100
+++ fw.patched/www/config/cp_maintenance.cgi	2018-01-09 11:28:06.092865645 +0100
@@ -347,109 +347,13 @@
                                 puts "\${dialogSettingsCMLblActualSoftwareVersion}"
                             }
                             table_data {
-                                puts "$cur_version"
-                            }
-                        }
-                        table_row {
-                            table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsCMLblAvailableSoftwareVersion}"
-                            }
-                            table_data {id="availableSWVersion"} {
-                                # This doesn´t work properly
-                                # puts [iframe "$REMOTE_FIRMWARE_SCRIPT?cmd=check_version&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2" marginheight=0 marginwidth=0 frameborder=0 width=100 height=20 {scrolling="no"} ]
-                                # The available version will be set further down with "jQuery('#availableSWVersion').html(homematic.com.getLatestVersion());"
-                            }
-                        }
-                        table_row {
-                            table_data {align="left"} {colspan="3"} {
-                                #puts "[bold "Software-Update durchführen"]"
-                                puts "<b>\${dialogSettingsCMLblPerformSoftwareUpdate}</b>"
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep1}"
-                            }
-                            table_data {
-                                division {class="popupControls CLASS20905"} {
-                                    table {
-                                        table_row {
-                                            table_data {
-                                                division {class="CLASS20908" style="display: none"} {id="btnFwDownload"} {} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {}
-                                                division {class="CLASS20908"}  "onClick=\"showCCULicense();\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep2}"
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {colspan="2"} {
-                                form "$env(SCRIPT_NAME)?sid=$sid" name=firmware_form {target=firmware_upload_iframe} enctype=multipart/form-data method=post {
-                                    export action=firmware_upload
-                                    export downloadOnly=$downloadOnly
-                                    file_button firmware_file size=30 maxlength=1000000
-                                }
-                                puts {<iframe name="firmware_upload_iframe" style="display: none;"></iframe>}
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep3}"
-                            }
-                            table_data {
-                                division {class="popupControls CLASS20905"} {
-                                    table {
-                                        table_row {
-                                            table_data {
-                                                division {class="CLASS20919"} {onClick="document.firmware_form.submit();showUserHint();"} {
-                                                  puts "\${dialogSettingsCMBtnPerformSoftwareUpdateUpload}"
-                                                }
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {colspan="2"} {class="CLASS20920"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep4}"
+                                puts "@@@pivccu_version@@@"
                             }
                         }
                     }
                 }
-                table_data {align="left"} {class="CLASS20921"} {
-                    puts "\${dialogSettingsCMHintSoftwareUpdate1}"
-                    number_list {class="j_noForcedUpdate"} {
-                        li {
-                          ${dialogSettingsCMHintSoftwareUpdate2}                        }
-                        li {
-                             ${dialogSettingsCMHintSoftwareUpdate3}
-                        }
-                        li {
-                             ${dialogSettingsCMHintSoftwareUpdate3a}
-                        }
-                        set bat_level [get_bat_level]
-                        if {$bat_level < 50} {
-                            set msg " \${dialogSettingsCMHintSoftwareUpdate4a} $bat_level%. "
-                            append msg  \${dialogSettingsCMHintSoftwareUpdate4b}
-                            li $msg
-                        }
-                    }
-
-                    division {class="j_forcedUpdate" style="padding:10px;"} {
-                      puts "<br/>\${dialogSettingsCMHintSoftwareUpdate3}"
-                    }
+                table_data {align="left"} {class="CLASS20904"} {
+                    puts "\${dialogSettingspiVCCUDonateHint}"
                 }
             }
             table_row {class="CLASS20902 j_noForcedUpdate j_fwUpdateOnly"} {
@@ -790,10 +694,6 @@
     cgi_javascript {
       puts "translatePage('#messagebox');"
       puts "jQuery('#messagebox').show();"
-      puts "var latestVersion = translateKey('lblAvailableFirmwareVersionNotKnown');"
-      puts "latestVersion = homematic.com.getLatestVersion();"
-      puts "if (latestVersion == undefined || isNaN(parseInt(latestVersion.split('.')\[0\]))) {latestVersion = translateKey('lblAvailableFirmwareVersionNotKnown');}"
-      puts "jQuery('#availableSWVersion').html(latestVersion);"
 
       # Hide all elements with class j_noForcedUpdate when an update is enforced
       puts "if (forceUpdate) {"
diff -ruN fw.orig/www/config/cp_security.cgi fw.patched/www/config/cp_security.cgi
--- fw.orig/www/config/cp_security.cgi	2018-01-09 11:17:52.752859308 +0100
+++ fw.patched/www/config/cp_security.cgi	2018-01-09 11:22:24.330632351 +0100
@@ -202,8 +202,8 @@
         # exec /usr/sbin/ubiattach -p /dev/mtd6
         # exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
         # exec mount /usr/local
-        exec touch /var/doFactoryReset
-        exec kill -SIGQUIT 1
+        eval [list exec rm -rf] [glob /usr/local/*]
+        exec /sbin/reboot
     }]} {
 
       # TWIST-22
@@ -476,12 +476,7 @@
     }
     
   if { "false" == $ccu1_backup } {  # backup for version >= 2
-    exec umount /usr/local
-          exec /usr/sbin/ubidetach -p /dev/mtd6
-          exec /usr/sbin/ubiformat /dev/mtd6 -y
-          exec /usr/sbin/ubiattach -p /dev/mtd6
-          exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
-          exec mount /usr/local
+    eval [list exec rm -rf] [glob /usr/local/*]
 
     if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
       # set msg "Beim Einspielen des Systembackups ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut. "
@@ -568,8 +563,6 @@
   }
   
   if { "false" == $backuperror } {
-        exec mount -o remount,ro /usr/local
-        exec mount -o remount,rw /usr/local
         division {class="popupTitle"} {
             puts "\${dialogSettingsSecurityMessageSysBackupRestartSystemTitle}"
         }
diff -ruN fw.orig/www/config/cp_software.cgi fw.patched/www/config/cp_software.cgi
--- fw.orig/www/config/cp_software.cgi	2018-01-09 11:17:52.748859282 +0100
+++ fw.patched/www/config/cp_software.cgi	2018-01-09 11:22:24.330632351 +0100
@@ -507,7 +507,7 @@
     if {[isOldCCU]} {
         exec /sbin/init -q
     } else {
-        exec /bin/kill -SIGQUIT 1
+        exec /sbin/reboot
     }
 }
 
diff -ruN fw.orig/www/webui/js/lang/de/translate.lang.js fw.patched/www/webui/js/lang/de/translate.lang.js
--- fw.orig/www/webui/js/lang/de/translate.lang.js	2018-01-09 11:17:52.584858212 +0100
+++ fw.patched/www/webui/js/lang/de/translate.lang.js	2018-01-09 11:22:24.330632351 +0100
@@ -445,7 +445,8 @@
     "dialogSettingsCMTDCCURestart" : HMIdentifier.de.CCUShortName + "<br/>Neustart",
     "dialogSettingsCMTDCCUSafeMode" : "Abgesicherter<br/>Modus",
     "dialogSettingsCMTDErrorProtocol" : "Fehler-<br/>protokoll",
-    "dialogSettingsCMLblActualSoftwareVersion" : "Aktuelle Software-Version:",
+    "dialogSettingsCMLblActualSoftwareVersion" : "Aktuelle piVCCU Version:",
+    "dialogSettingspiVCCUDonateHint" : "Bitte unterst%FCtzen Sie die Entwicklung von piVCCU mit einer Spende via <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=KJ3UWNDMXLJKU' rel='nofollow'>PayPal</a>.",
     "dialogSettingsCMLblAvailableSoftwareVersion" : "Verf%FCgbare Software-Version:",
     "dialogSettingsCMLblPerformSoftwareUpdate" : "Software-Update durchf%FChren",
     "dialogSettingsCMLblPerformSoftwareUpdateStep1" : "Schritt 1: Neue Software herunterladen",
diff -ruN fw.orig/www/webui/js/lang/en/translate.lang.js fw.patched/www/webui/js/lang/en/translate.lang.js
--- fw.orig/www/webui/js/lang/en/translate.lang.js	2018-01-09 11:17:52.584858212 +0100
+++ fw.patched/www/webui/js/lang/en/translate.lang.js	2018-01-09 11:22:24.330632351 +0100
@@ -445,7 +445,8 @@
     "dialogSettingsCMTDCCURestart" : HMIdentifier.en.CCUShortName + "<br/>restart",
     "dialogSettingsCMTDCCUSafeMode" : "Safe<br/>mode",
     "dialogSettingsCMTDErrorProtocol" : "Error<br/>log",
-    "dialogSettingsCMLblActualSoftwareVersion" : "Current software version:",
+    "dialogSettingsCMLblActualSoftwareVersion" : "Current piVCCU version:",
+    "dialogSettingspiVCCUDonateHint" : "Please support the development of piVCCU with a donation via <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=KJ3UWNDMXLJKU' rel='nofollow'>PayPal</a>.",
     "dialogSettingsCMLblAvailableSoftwareVersion" : "Available software version:",
     "dialogSettingsCMLblPerformSoftwareUpdate" : "Perform software update",
     "dialogSettingsCMLblPerformSoftwareUpdateStep1" : "Step 1: Download new software",
